concordium-sdk/src/main/java/com/concordium/sdk/cis2/SerializationUtils.java

```diff
@@ -0,0 +1,297 @@
+        val urlLength = UInt16.from(resultBuffer.getShort()).getValue();
+        resultBuffer.order(ByteOrder.BIG_ENDIAN);
+        val urlBytes = new byte[urlLength];
+        resultBuffer.get(urlBytes);
+        val hasChecksum = resultBuffer.get() != 0;
+        byte[] checksumBuffer = null;
+        if (hasChecksum) {
+            checksumBuffer = new byte[32];
+            resultBuffer.get(checksumBuffer);
+        }
+        return new TokenMetadata(new URL(new String(urlBytes, StandardCharsets.UTF_8)), checksumBuffer);
+    }
+
+    @SneakyThrows
+    static byte[] serializeTokenId(TokenId tokenId) {
+        // size of token + serialized token id.
+        val tokenSize = tokenId.getSize();
+        val buffer = ByteBuffer.allocate(1 + tokenSize);
+        buffer.put((byte) tokenSize);
+        if (tokenSize != 0) {
+            buffer.put(tokenId.getBytes());
+        }
+        return buffer.array();
+    }
+
+    static byte[] serializeAddress(AbstractAddress address) {
+        if (address instanceof AccountAddress) {
+            val accountAddress = (AccountAddress) address;
+            val buffer = ByteBuffer.allocate(ACCOUNT_ADDRESS_SIZE);
+            buffer.put((byte) 0); // tag
+            buffer.put(accountAddress.getBytes());
+            return buffer.array();
+        } else if (address instanceof ContractAddress) {
+            ContractAddress contractAddress = (ContractAddress) address;
+            val buffer = ByteBuffer.allocate(CONTRACT_ADDRESS_SIZE);
+            buffer.put((byte) 1); // tag
+            // index and sub-index are stored as little endian.
+            buffer.order(ByteOrder.LITTLE_ENDIAN);
+            buffer.putLong(contractAddress.getIndex());
+            buffer.putLong(contractAddress.getSubIndex());
+            return buffer.array();
+        }
+        throw new IllegalArgumentException("AbstractAddress must be either an account address or contract address");
+    }
+
+    @SneakyThrows
+    public static Parameter serializeUpdateOperators(Map<AbstractAddress, Boolean> operatorUpdates) {
+        val bos = new ByteArrayOutputStream();
+        bos.write(UInt16.from(operatorUpdates.size()).getBytesLittleEndian());
+        for (AbstractAddress address : operatorUpdates.keySet()) {
+            bos.write((byte) (operatorUpdates.get(address) ? 1 : 0));
+            bos.write(SerializationUtils.serializeAddress(address));
+        }
+        return Parameter.from(bos.toByteArray());
+    }
+
+    private static int readUnsignedLeb128(ByteBuffer buffer) {
+        int result = 0;
+        int cur;
+        int count = 0;
+        do {
+            cur = buffer.get() & 0xff;
+            result |= (cur & 0x7f) << (count * 7);
+            count++;
+        } while (((cur & 0x80) == 0x80) && count < 5);
+        if ((cur & 0x80) == 0x80) {
+            throw new IllegalArgumentException("invalid LEB128 encoding");
+        }
+        return result;
+    }
```

Comment:
This does not look correct. Why do we only look at 5 bytes?

Also why are you parsing this into an int, amounts can be much bigger.

Reply:
I see that now. Fixed.